### ISSUE TITLE
Index out of range error

### ISSUE THREAD

**Issue by Checksum** [lafin/brief#1]
Hi, 

Thanks for this as I've been looking for something similar for a while now! I tried to run the example at https://github.com/lafin/fast-brief with a bunch of different images, and I run into this error intermittently:

```
PANIC: runtime error: index out of range
/mnt/d/projects/src/vendor/github.com/lafin/brief/brief.go:139 +0x25
```

It keeps throwing up an error on https://github.com/lafin/brief/blob/master/brief.go#L139. So basically `offsets[offsetsPosition-2]+w` or `offsets[offsetsPosition-1]+w` either `< 0` or `> len(pixels)`. Haven't been able to determine what exactly causes this.

Would appreciate if you have any thoughts on this.

Thanks!

**Comment by lafin** [comment]
Can you provide me samples of image which caused the error?

**Comment by Checksum** [comment]
Sorry for the delayed response, I kind of moved away from this for a while. What I'm trying to do is get two images from the user and check if they are similar. For example: 
![jeans](https://user-images.githubusercontent.com/91416/37470559-e047ff82-28bb-11e8-9d32-7c2ef06124b4.jpg)

Here are the pre-processing steps:

1. Converted to PNG (is this required?)
2. Converted to grayscale

I've been able to run the images that you have in the `fast-brief` repo successfully, but always fails on my images. I've tried stripping out EXIF data from my images as well, but no avail.

Is there a specific format in which the pixel array is expected to be in? I'm sure I'm missing something important, but I have no idea! Any thoughts on this would be appreciated greatly!
